cocotb/share/lib/simulator/simulatormodule.cpp

```diff
@@ -921,138 +921,164 @@ PyMODINIT_FUNC PyInit_simulator(void)
     return simulator;
 }
 
+/* NOTE: in the following docstrings we are specifying the parameters twice, but this is necessary.
+ * The first docstring before the long '--' line specifies the __text_signature__ that is used
+ * by the help() function. And the second after the '--' line contains type annotations used by
+ * Sphinx for generating documentation because type annotations are not supported in
```

Comment:
```suggestion
 * the `autodoc_docstring_signature` setting of sphinx.ext.autodoc for generating documentation because type annotations are not supported in
```

(reference [here](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#confval-autodoc_docstring_signature))

Modulo line-wrapping which is too hard to do in github comments.